src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -151,6 +152,17 @@ internal static RuntimeGraph GetRuntimeGraph(IEnumerable targetFrameworks)
             return tfi;
         }
 
+        internal static NuGetFramework GetTargetFramework(IVsProjectProperties properties)
+        {
+            var targetFrameworkIdentifier = GetPropertyValueOrNull(properties, ProjectBuildProperties.TargetFrameworkIdentifier) ?? string.Empty;
+            var targetFrameworkVersion = GetPropertyValueOrNull(properties, ProjectBuildProperties.TargetFrameworkVersion) ?? string.Empty;
```

Comment:
Do you need to provide empty strings? I'm not sure there'll be a big difference in behavior of null vs empty.

Reply:
Before `NuGetFramework.ParseComponents` was added, I was calling the `NuGetFramework` constructor, which did throw when null. However, `NuGetFramework.ParseComponents` doesn't, so it's no longer needed.